src/app/components/FilterButtons/index.tsx

```diff
@@ -0,0 +1,42 @@
+import Box from '@mui/material/Box'
+import Chip from '@mui/material/Chip'
+import Typography from '@mui/material/Typography'
+import { COLORS } from '../../../styles/theme/colors'
+
+type FilterButtonsProps<T> = {
+  options: { label: string; value: T }[]
+  onSelect: (value: T) => void
+  value?: T
+}
+
+export const FilterButtons = <T,>({ options, onSelect, value }: FilterButtonsProps<T>) => {
+  return (
+    <Box sx={{ display: 'inline-flex' }}>
+      {options.map(option => {
+        const selected = option.value === value
+        return (
+          <Chip
+            key={String(option.value)}
```

Comment:
This is a bit risky if `T` can be anything, including objects. Restrict it with `T extends string`?